javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -426,7 +434,11 @@ public final void write(ClientMethod clientMethod, JavaType typeBlock) {
                 break;
 
             case LongRunningBeginAsync:
-                generateLongRunningBeginAsync(clientMethod, typeBlock, restAPIMethod, settings);
+                if (settings.isLowLevelClient()) {
+                    generateProtocolLongRunningBeginAsync(clientMethod, typeBlock, restAPIMethod, settings);
+                } else {
+                    generateLongRunningBeginAsync(clientMethod, typeBlock, restAPIMethod, settings);
```

Comment:
Please check if you need to handle the same issue for vanilla as well. Context need to be passed to the Strategy as well (maybe in `generateLongRunningBeginAsync`).

Reply:
ok, the test is used for validate context is passed in  correctly to the policy.

Reply:
I mean you do not need to add more test for vanilla, but make sure generated code is correct with `context` set to `Strategy` instances.